src/Tizen.NUI/src/internal/Layouting/RelativeLayout.cs

```diff
@@ -0,0 +1,270 @@
+                {
+                    return false;
+                }
+            }
+            else if (space.start == 0.0)
+            {
+                if (viewPosion + viewSize > layoutSize)
+                    return false;
+            }
+            else if (space.start == layoutSize)
+            {
+                if (viewPosion < 0)
+                    return false;
+            }
+            else if (viewPosion < 0 || viewPosion + viewSize > layoutSize)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        private bool ValidateRelatives(in float layoutSize, HashSet<View> fixedRelativeSet, GetRelative GetRelative)
+        {
+            foreach (LayoutItem childLayout in LayoutChildren.Where(item => item?.Owner != null))
+            {
+                Relative relative = GetRelative(childLayout.Owner, layoutSize);
+                if (!ValidateRelative(relative, layoutSize, fixedRelativeSet.Contains(childLayout.Owner)))
+                    return false;
+            }
+
+            return true;
+        }
+
+        private HashSet<View> GetFixedRelativeSet(in float layoutSize, GetRelative GetRelative)
+        {
+            HashSet<View> ignoreSet = new HashSet<View>();
+
+            foreach (LayoutItem childLayout in LayoutChildren.Where(item => item?.Owner != null))
+            {
+                Relative relative = GetRelative(childLayout.Owner, layoutSize);
+                if (relative.spaceGeometry.Size > float.Epsilon
+                    && relative.spaceGeometry.Size < relative.viewGeometry.Size)
+                {
+                    ignoreSet.Add(childLayout.Owner);
+                }
+            }
+
+            return ignoreSet;
+        }
+
+        private Relative GetHorizontalRelative(View view, in float layoutSize) => CalculateRelative(view, layoutSize, 0, RelativeHorizontalPropertyGetters);
+
+        private Relative GetVerticalRelative(View view, in float layoutSize) => CalculateRelative(view, layoutSize, 0, RelativeVerticalPropertyGetters);
+
+        private Relative CalculateRelative(View view, in float layoutSize, int depth, RelativePropertyGetters propertyGetters)
+        {
+            if (propertyGetters.RelativeCache.TryGetValue(view, out Relative cache))
+                return cache;
+
+            depth++;
+            if (depth > LayoutChildren.Count)
+            {
+                throw new InvalidOperationException("Circular dependency detected in RelativeLayout.");
+            }
+            Relative cacheFrom;
+            Relative cacheTo;
+            float RelativeOffsetFrom = propertyGetters.GetRelativeOffsetFrom(view);
+            float RelativeOffsetTo = propertyGetters.GetRelativeOffsetTo(view);
+            float viewSize = propertyGetters.GetMeasuredSize(view.Layout);
+
+            if (propertyGetters.GetTargetFrom(view) is View targetFrom && targetFrom != Owner)
```

Comment:
Some `if` conditionals of this file miss braces on separate lines. Also, after `else` keyword, the following clause must always be surrounded by braces.
But, I'm not sure whether the braces are necessary to NUI `if` statement. Could you give your opinion?

```suggestion
            if (propertyGetters.GetTargetFrom(view) is View targetFrom && targetFrom != Owner)
            {
                cacheFrom = CalculateRelative(targetFrom, layoutSize, depth, propertyGetters);
            }
```

Reply:
I don't have a strict rule for this case. i will add braces, if you want.

Reply:
Thanks :)